### PR TITLE
custom addr and fix reset timing issue

### PR DESCRIPTION
**Pull request by dragazo**
This adds a param that can be used to set a custom i2c address.

Also fixes a timing issue where the reset right before reading calibration data can result in the first few bytes of calibration data being zero (which permanently throws off all readings since there's no public api to re-read the calibration data).

### CODE REVIEW

**Comment by elpiel** [comment]
@dragazo have you been able to make this sensor work?
I get a status of temp. and pressure not ready, even though I've set every possible configuration and values don't change over time.

@ibutra would you be interested to allow other maintainers to the repo?
We at @aerorust need it and would be nice to add support for traits from newer embedded-hal, including async.
I couldn't find any contact details to reach out to you and issues are disabled on this repo...

**Comment by ibutra** [comment]
@dragazo Sry I overlooked this pull request, but to be honest I haven't used this crate and or the sensor in a long time so I don't know if I am the right one to review it.

So @elpiel your suggesting to have other maintainers is just fine for me.

**Comment by elpiel** [comment]
Yes, I would like to take over the crate inside the @aerorust org. repo. 

There are also a few fixes amd improvements I would like to apply.

**Comment by elpiel** [comment]
@ibutra would you like to give me access to crates.io?
If you give me access to the repo as maintainer I can move it to AeroRust org.
FYI: I will leave you as original author of the crate and do not plan to remove you in any way.

**Comment by dragazo** [comment]
@elpiel I got the sensor working and reading correctly calibrated data. But my use case didn't require checking the ready flag, so that part may still be broken.

**Comment by elpiel** [comment]
@dragazo I keep getting the same temperature and pressure when I read the values that's why I assume it's not working correctly.

**Comment by dragazo** [comment]
@elpiel Well mine is working, but I think the default power setting is on-demand, but not via the normal functions that get readings. Here is my [initialization code](https://github.com/dragazo/netsblox-vm-esp32/blob/d00a1da39e97b1e9af18c33047712c4e9581e0f1/src/platform.rs#L690-L703) and [polling code](https://github.com/dragazo/netsblox-vm-esp32/blob/d00a1da39e97b1e9af18c33047712c4e9581e0f1/src/platform.rs#L945-L951). This works in the current version of this PR.

**Comment by ibutra** [comment]
I made you both collaborators. Whether you want to move this repository somewhere else I would leave to you two. 
I also invited you both to the crates.io as owner

**Comment by elpiel** [comment]
Awesome! Thank you, will try to create some issue for people and me to take a look.

**Comment by elpiel** [comment]
Hmmm.. ok, so I did try the normal mode and commented out all other settings fn calls - it started working :tada:
I tried adding back the oversampling setting and that's where I started getting the `pressure data ready? false; temperature data ready? false`

I'll investigate the other setting as well and open bug issues accordingly to my findings.

**Comment by elpiel** [comment]
PowerMode::Forced also results in the same behaviour although I do get initially a read state `true` for both sensors and then it drops to `false`.
I will familiarize with the datasheet first to get a better understanding what to expect and when.

**Comment by elpiel** [comment]
Ok, I figured it out.
Accordingly to the datasheet, when using Force mode you must set the mode before each readout as the chip goes back to sleep.

**Comment by elpiel** [comment]
@dragazo accordingly to the datasheet for the i2c address:
> is 111011x. The 6 MSB bits are fixed. The last bit is changeable by SDO value and can be changed
during operation. Connecting SDO to GND results in slave address 1110110 (0x76); connection it to VDDIO results in slave
address 1110111 (0x77)

What do you think about adding an enum, instead of allowing any i2c address. I've seen this practice in other driver crates.

We can apply the change in this PR or change it after the merge but wanted to ask for opinions first.

**Comment by dragazo** [comment]
Personally I found that practice in other crates a bit annoying since under the hood it really is just a `u8`. For instance my use case actually requires the address be set dynamically (not at compile time, but still at startup time) by the user in a json config. So the rigid enum approach means I'd have to actually parse the `u8` into the enum and fail if it's not either of those two options. But maybe we could do a compromise like:

```rust
#[derive(Debug, Clone, Copy)]
pub struct Addr {
    Primary,
    Secondary,
    Dynamic(u8),
}
impl Addr {
    pub fn get(self) -> u8 {
        match self {
            Primary => 0x76,
            Secondary => 0x77,
            Dynamic(x) => x,
        }
    }
}
```

**Comment by elpiel** [comment]
The reasoning about the enum is that you can't have any address for the sensor, the valid values are only 2 accordingly to the datasheet.

Either way you set the address on initialising the struct so the only thing we can do is add an `impl TryFrom<u8>` that would help with your case.

> The 7-bit device address is 111011x. The 6 MSB bits are fixed. The last bit is changeable by SDO value and can be changed
during operation. Connecting SDO to GND results in slave address 1110110 (0x76); connection it to VDDIO results in slave
address 1110111 (0x77), which is the same as BMP180’s I²C address. The SDO pin cannot be left floating; if left floating, the
I²C address will be undefined

Not really sure what you mean by dynamic in this case.

**Comment by elpiel** [comment]
And we can always add enumeration for the enum:
```
#[derive(Debug, Clone, Copy)]
#[repr(u8)]
pub enum Addr {
    Primary => 0x76,
    Secondary = 0x77,
}
```

then you can do
```
let addr = Address::Primary as u8;
assert_eq!(addr, 0x76);

```

**Comment by dragazo** [comment]
By dynamic I just mean it's controlled by some other logic (whoever provides the wrapped `u8` value). Although the datasheet only says those two addresses are valid, there's no reason they couldn't release a new version of the chip later that has extra addresses, so I just prefer allowing the user to directly provide a `u8` instead of trying to do it for them indirectly. I also like the `[repr(u8)]` approach - I was actually considering just doing that instead. As long as we show an example in the docs it should be fine.

**Comment by dragazo** [comment]
I just added the `repr(u8)` version.

**Comment by elpiel** [comment]
> there's no reason they couldn't release a new version of the chip later that has extra addresses,
Yes, I agree that flexibility is good, however, we can always release a new version of the crate with a breaking change and bump the minor or major version.

Thanks for adding it!

**Review by elpiel — APPROVED** [review]

**Pull request merged**
